tests/frontend/specs/select_formatting_buttons.js

```diff
@@ -125,33 +130,28 @@ describe('select formatting buttons when selection has style applied', function
 
   context('when user applies a style and the selection does not change', function () {
     const style = STYLES[0]; // italic
-    before(function () {
-      applyStyleOnLine(style, FIRST_LINE);
-    });
-
-    // clean the style applied
-    after(function () {
-      applyStyleOnLine(style, FIRST_LINE);
-    });
 
     it('selects the style button', function (done) {
+      applyStyleOnLine(style, FIRST_LINE);
+      helper.waitForPromise(() => isButtonSelected(style) === true);
```

Comment:
needs an await, or it will run the code below helper.waitForPromise even if the Promise is rejected (afaik this would result in a green test, with an printed error in case the Promise is rejected)